template-dockerfile.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+COMPONENT="$1"
+BASE_DOCKERFILE=Dockerfile
+go_mod_file=${COMPONENT}/go.mod
+
+# Copy base Dockerfile to create component-specific Dockerfile
+# eg: featuregates/controller -> featuregates.controller.Dockerfile
+COMPONENT_DOCKERFILE=${COMPONENT////.}.Dockerfile
+cp ${BASE_DOCKERFILE} ${COMPONENT_DOCKERFILE}
+
+# Get relative references from replace directive statements
+# Only those replace statements that start with '.'
+replace_refs=($(awk -F ' => ' '{ if($2~/^\./) print $2 }' ${go_mod_file}))
+
+if [ ${#replace_refs[@]} -eq 0 ]; then
```

Comment:
Do we need this special case?

Reply:
Its just an early exit condition to avoid additional computations if no valid replace statements are found. Since the awk command in line 46 will read the entire file to find the COPY pattern and replace nothing.